qsiprep/tests/test_utils_grouping.py

```diff
@@ -386,204 +374,279 @@ def test_group_dwi_scans_with_complex_b0fields(tmpdir):
+    but with IntendedFors using relative paths instead of B0Field* fields.
+    """
+    layout, subject_data = complex_relpaths_dataset
 
     scan_groups, _ = grouping.group_dwi_scans(
         layout=layout,
         subject_data=subject_data,
-        using_fsl=False,
-        combine_scans=False,
-        ignore_fieldmaps=False,
-        concatenate_distortion_groups=False,
+        combine_scans=combine_scans,
+        ignore_fieldmaps=ignore_fieldmaps,
     )
-    expected = [
-        {
-            'concatenated_bids_name': 'sub-01_dir-AP_run-1',
-            'dwi_series': ['sub-01_dir-AP_run-1_dwi.nii.gz'],
-            'dwi_series_pedir': 'j-',
-            'fieldmap_info': {
-                'epi': ['sub-01_dir-PA_epi.nii.gz'],
-                'suffix': 'epi',
-            },
-        },
-        {
-            'concatenated_bids_name': 'sub-01_dir-AP_run-2',
-            'dwi_series': ['sub-01_dir-AP_run-2_dwi.nii.gz'],
-            'dwi_series_pedir': 'j-',
-            'fieldmap_info': {
-                'epi': ['sub-01_dir-PA_epi.nii.gz'],
-                'suffix': 'epi',
-            },
-        },
-        {
-            'concatenated_bids_name': 'sub-01_dir-PA',
-            'dwi_series': ['sub-01_dir-PA_dwi.nii.gz'],
-            'dwi_series_pedir': 'j',
-            'fieldmap_info': {
-                'epi': ['sub-01_dir-AP_epi.nii.gz'],
-                'suffix': 'epi',
-            },
-        },
-    ]
     check_expected(scan_groups, expected)
 
-    scan_groups, _ = grouping.group_dwi_scans(
-        layout=layout,
-        subject_data=subject_data,
-        using_fsl=True,
-        combine_scans=True,
-        ignore_fieldmaps=True,
-        concatenate_distortion_groups=False,
-    )
-    expected = [
-        {
-            'concatenated_bids_name': 'sub-01_dir-AP',
-            'dwi_series': [
-                'sub-01_dir-AP_run-1_dwi.nii.gz',
-                'sub-01_dir-AP_run-2_dwi.nii.gz',
+
+@pytest.mark.parametrize(
+    ('combine_scans', 'ignore_fieldmaps', 'expected'),
+    [
+        # Test case 1: combine_scans=True, ignore_fieldmaps=False
+        (
+            True,
+            False,
+            [
+                {
+                    'concatenated_bids_name': 'sub-01',
+                    'dwi_series': ['sub-01_dir-LR_dwi.nii.gz'],
+                    'dwi_series_pedir': 'i',
+                    'fieldmap_info': {
+                        'rpe_series': ['sub-01_dir-RL_dwi.nii.gz'],
+                        'suffix': 'rpe_series',
+                    },
+                },
+                {
+                    'concatenated_bids_name': 'sub-01',
+                    'dwi_series': ['sub-01_dir-PA_dwi.nii.gz'],
+                    'dwi_series_pedir': 'j',
+                    'fieldmap_info': {
+                        'rpe_series': ['sub-01_dir-AP_dwi.nii.gz'],
+                        'suffix': 'rpe_series',
+                    },
+                },
             ],
```

Comment:
It looks like all of the different PEDs aren't being combined. Also, these are conflicting concatenated BIDS names (`sub-01_dir-AP` + `sub-01_dir-PA` --> `sub-01` and `sub-01_dir-RL` + `sub-01_dir-LR` --> `sub-01`)

Reply:
I think if we're not using DRBUDDI these should all go into sub-01 because TOPUP will estimate a single field from them all. I'm not sure what the bidsiest way would be to name the outputs for DRBUDDI. Maybe `sub-01_dir-APPA` and `sub-01_dir-RLLR`?

Reply:
That's a hard one. Derivatives are still largely unchecked, but theoretically the `dir` entity corresponds to the `PhaseEncodingDirection` metadata field, meaning that the validator _should_ check that `PhaseEncodingDirection` is present when `dir` is, but it can't be since there is no PED in the combined files.